frame/bridge/ethereum/relay/src/lib.rs

```diff
@@ -1022,43 +1023,31 @@ impl<T: Send + Sync + Config> SignedExtension for CheckEthereumRelayHeaderParcel
 	fn validate(
 		&self,
 		_: &Self::AccountId,
-		_call: &Self::Call,
+		call: &Self::Call,
 		_: &DispatchInfoOf<Self::Call>,
 		_: usize,
 	) -> TransactionValidity {
-		// TODO: pre-verify
-		// if let Some(Call::submit_proposal(ref proposal)) = call.is_sub_type() {
-		// 	if let Some(proposed_header_thing) = proposal.get(0) {
-		// 		for existed_proposal in
-		// 			T::RelayerGame::proposals_of_game(proposed_header_thing.header.number)
-		// 		{
-		// 			if existed_proposal
-		// 				.bonded_proposal
-		// 				.iter()
-		// 				.zip(proposal.iter())
-		// 				.all(
-		// 					|(
-		// 						(
-		// 							_,
-		// 							EthereumHeaderThing {
-		// 								header: header_a,
-		// 								mmr_root: mmr_root_a,
-		// 							},
-		// 						),
-		// 						EthereumHeaderThingWithProof {
-		// 							header: header_b,
-		// 							mmr_root: mmr_root_b,
-		// 							..
-		// 						},
-		// 					)| header_a == header_b && mmr_root_a == mmr_root_b,
-		// 				) {
-		// 				return InvalidTransaction::Custom(<Error<T>>::AffirmationInv.as_u8()).into();
-		// 			}
-		// 		}
-		// 	}
-		// }
-
-		Ok(ValidTransaction::default())
+		match call.is_sub_type() {
+			Some(Call::affirm(ref relay_header_parcel, _)) => {
+				ensure!(
+					T::RelayerGame::get_affirmed_relay_header_parcels(&RelayAffirmationId {
+						game_id: relay_header_parcel.header.number,
+						round: 0,
+						index: 0,
+					})
+					.is_none(),
```

Comment:
Two relayers may affirm two different headers? It is not enough to pre validate only check the `RelayAffirmationId` as key duplication, checking `relay_header_parcel` is also required.

Reply:
No dispute API on bridger side. There is only one header now.